src/main/java/com/googlecode/lanterna/gui2/AbstractListBox.java

```diff
@@ -65,6 +70,114 @@ protected AbstractListBox(TerminalSize size) {
         setListItemRenderer(createDefaultListItemRenderer());
     }
 
+    @Override
+    public void setIsWithinScrollPanel(boolean isWithinScrollPanel) {
+        this.isWithinScrollPanel = isWithinScrollPanel;
+    }
+    
+    @Override
+    public boolean isVerticalScrollCapable() {
+        return true;
+    }
+    
+    @Override
+    public boolean isVerticalScrollVisible() {
+        int componentHeight = getSize().getRows();
+        int itemCount = getItems().size();
+        
+        return itemCount > componentHeight;
+    }
+    
+    @Override
+    public int getVerticalScrollMaximum() {
+        return getItems().size();
+    }
+    
+    @Override
+    public int getVerticalScrollPosition() {
+        return - scrollOffset.getRow();
```

Comment:
Is the `-` a typo?

Reply:
Not a typo, but redoing this part.
The ScrollableBox interface kinda became too cumbersome, so taking a draft 2 approach.